src/components/common/YourCompany/ChangeBusinessType.vue

```diff
@@ -356,39 +292,32 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
     this.confirmArticles = false
   }
 
-  /** Entity Options. */
-  readonly entityTypeOptions = [
-    {
-      value: 'BC',
-      SHORT_DESC: 'BC Limited Company',
-      text: 'BC Limited Company'
-    },
-    {
-      value: 'BEN',
-      SHORT_DESC: 'BC Benefit Company',
-      text: 'BC Benefit Company'
-    }
-  ]
-
   /** Verify New Business name. */
   get isNewName (): boolean {
     return this.getNameRequestLegalName &&
       (this.getNameRequestLegalName !== this.getEntitySnapshot?.businessInfo?.legalName)
   }
 
-  /** Check if current entity selection is a Benefit Company */
-  get isBenefit (): boolean {
-    return (this.selectedEntityType === CorpTypeCd.BENEFIT_COMPANY)
-  }
-
   /** Type change helper information */
   get typeChangeInfo (): string {
     return this.getResource.changeData?.typeChangeInfo
   }
 
+  /** Entity type options based on the company type */
+  get entityTypeOptions (): EntityTypeOption[] {
+    return this.getResource.changeData?.entityTypeOptions
```

Comment:
If changeData is falsy then this returns an undefined.

Should this return an empty array if changeData is falsy? Or can the code that uses this handle an undefined value?

Reply:
Probably could write 

`this.getResource.changeData?.entityTypeOptions || []`